src/ert/_c_wrappers/enkf/ert_workflow_list.py

```diff
@@ -218,31 +127,109 @@ def __eq__(self, other):
         for name_self, name_other in zip(
             sorted(self.getJobNames()), sorted(other.getJobNames())
         ):
-            job_self = self.getJob(name_self)
-            job_other = other.getJob(name_other)
-            if job_self != job_other:
+            if self.getJob(name_self) != other.getJob(name_other):
                 return False
 
-        if self.getWorkflowNames() != other.getWorkflowNames():
+        if set(self.getWorkflowNames()) != set(other.getWorkflowNames()):
             return False
 
         for name_self, name_other in zip(
-            self.getWorkflowNames(), other.getWorkflowNames()
+            sorted(self.getWorkflowNames()), sorted(other.getWorkflowNames())
         ):
             if self[name_self] != other[name_other]:
                 return False
 
         if self._hook_workflows != other._hook_workflows:
             return False
 
+        for mode in HookRuntime.enums():
+            for name_self, name_other in zip(
+                sorted(self.get_workflows_hooked_at(mode)),
+                sorted(other.get_workflows_hooked_at(mode)),
+            ):
+                if name_self != name_other:
+                    return False
+                if name_self.src_file != name_other.src_file:
+                    return False
+
         return True
 
+    def _handle_config_content_dict(self, content_dict):
+        for workflow in content_dict.get(ConfigKeys.LOAD_WORKFLOW_JOB, []):
+            self._workflow_job_info.append(workflow)
+
+        for workflow_dir in content_dict.get(ConfigKeys.WORKFLOW_JOB_DIRECTORY, []):
+            self._workflow_job_dir_info.append(workflow_dir)
+
+        for name, mode in content_dict.get(ConfigKeys.HOOK_WORKFLOW_KEY, []):
+            self._hook_workflow_info.append([name, mode])
+
+        for workflow in content_dict.get(ConfigKeys.LOAD_WORKFLOW, []):
+            self._workflow_info.append(workflow)
+
+    def _handle_config_dict(self, config_dict):
+        for job in config_dict.get(ConfigKeys.LOAD_WORKFLOW_JOB, []):
+            self._workflow_job_info.append([job[ConfigKeys.PATH], job[ConfigKeys.NAME]])
+
+        for job_path in config_dict.get(ConfigKeys.WORKFLOW_JOB_DIRECTORY, []):
+            self._workflow_job_dir_info.append(job_path)
+
+        for hook in config_dict.get(ConfigKeys.HOOK_WORKFLOW_KEY, []):
+            self._hook_workflow_info.append(
+                [hook[ConfigKeys.NAME], hook[ConfigKeys.RUNMODE]]
```

Comment:
The difference between `_handle_config_content_dict` and `_handle_config_dict` is just the lookup of `ConfigKeys.Name`, `ConfigKeys.Path`, and `ConfigKeys.RUNMODE` are indices instead. Because there is no code using `_handle_config_dict` (except for tests). I suggest we change the expected input for that to be integer indices instead of strings so that we can have just one code path.

Reply:
After talking in person, we decided to start a new PR for this.